test-d/global-this.ts

```diff
@@ -0,0 +1,16 @@
+import {expectType} from 'tsd';
+import type {GlobalThis} from '../index';
+
+type ExtraProps = GlobalThis & {
+	readonly GLOBAL_TOKEN: string;
+};
+
+// Verify `globalThis` type objects can be casted with a type which extends `GlobalThis`.
+expectType<string>((globalThis as ExtraProps).GLOBAL_TOKEN);
+expectType<string>((window as ExtraProps).GLOBAL_TOKEN);
+expectType<string>((self as ExtraProps).GLOBAL_TOKEN);
+
+// Verify that non-`globalThis` type objects cannot be casted with a type which extends `GlobalThis`.
+declare function consume(): void;
+// @ts-expect-error
+consume(({something: 'value'}) as ExtraProps);
```

Comment:
How does this test work? By looking at it it seems that `@ts-expect-error` would simply catch the fact that you're passing an argument to a function that takes no arguments.

Reply:
Yeah. Head palm. That's the trouble with `@ts-expect-error`. It's too broad.

Thx for pointing this out. Fixed now.